distribution/lib/Standard/Base/0.0.0-dev/src/Data/Index_Sub_Range.enso

```diff
@@ -211,8 +211,8 @@ take_helper length at single_slice slice_ranges range = case range of
         trimmed = resolve_ranges indices length
         slice_ranges trimmed
     Index_Sub_Range.Sample count seed ->
-        rng = Random.new seed
-        indices_to_take = Random.random_indices length count rng
+        Random.set_seed seed
+        indices_to_take = Random.indices length count
```

Comment:
This isn't great.

Before, computing `Sample` did not have any noticeable side effects.

Now, it will affect the results of other computations, and will 'undo' the effects of `set_seed` - especially knowing that in 99% of the usecases it will keep the default `get_default_seed` which is changing on every recomputation.

----

Maybe we should still be able to create a custom Random generator instance with a custom seed - so that an operation like this one can compute its results without influencing others?

It would not be a user-facing API but an internal one.

Reply:
Very good point, there's no reason at all not to use instances within libraries.

Reply:
Done